storm-core/src/clj/backtype/storm/daemon/executor.clj

```diff
@@ -671,7 +689,19 @@
                                     user-context (:user-context task-data)
                                     sampler? (sampler)
                                     execute-sampler? (execute-sampler)
-                                    now (if (or sampler? execute-sampler?) (System/currentTimeMillis))]
+                                    now (if (or sampler? execute-sampler?) (System/currentTimeMillis))
+                                    receive-queue (:receive-queue executor-data)]
+                                (if (and ((:storm-conf executor-data) TOPOLOGY-BACKPRESSURE-ENABLE) 
+                                         (> (.population receive-queue) high-watermark) 
+                                         (not @(:backpressure executor-data)))
+                                  (do (reset! (:backpressure executor-data) true)
+                                      (log-debug "executor " (:executor-id executor-data) " is congested, set backpressure flag true")
+                                      (disruptor/notify-backpressure-checker (:backpressure-trigger (:worker executor-data)))))
+                                (if (and ((:storm-conf executor-data) TOPOLOGY-BACKPRESSURE-ENABLE) 
+                                         (< (.population receive-queue) low-watermark) 
+                                         @(:backpressure executor-data))
+                                  (do (reset! (:backpressure executor-data) false)
+                                      (disruptor/notify-backpressure-checker (:backpressure-trigger (:worker executor-data)))))
```

Comment:
I am a bit confused.  Why are we doing this as part of the metrics tick?  What does that have to do with back pressure at all?